pandas/tests/io/formats/test_to_csv.py

```diff
@@ -563,3 +563,23 @@ def test_to_csv_compression(self, compression_only,
             result = pd.read_csv(path, index_col=0,
                                  compression=read_compression)
             tm.assert_frame_equal(result, df)
+
+
+
+def test_csv_formatter_line_terminator_default(monkeypatch):
+    # see GH #23608
+    # ensure default line_terminator used is os.linesep
+    df = tm.makeDataFrame()
+
+    with monkeypatch.context() as m:
+        # fake default os.linesep to look like windows
+        m.setattr(os, 'linesep', '\r\n')
+        assert os.linesep == '\r\n'
+        # import within this context so it uses the patched os.linesep
+        from pandas.io.formats.csvs import CSVFormatter
+
+        formatter_with_argument = CSVFormatter(df, line_terminator='\r')
+        assert formatter_with_argument.line_terminator == '\r'
+
+        formatter_with_default = CSVFormatter(df)
```

Comment:
Does this need to be done within the context of the monkey patch? Seems more logical outside

Reply:
On master if you run this without patching the line terminator is set to \n by default even if you are on Windows. This is essentially the bug I'm trying to fix. If you don't patch here and expect the default to equal os.linesep the test will pass on \n platforms and fail on Windows.